ci: use Rook 1.19.3 and Ceph Tentacle

nixpanic opened the pull request:
Recent versions of Rook use Ceph-CSI Operator by default. There is no need to have Ceph-CSI deployed by Rook, as our CI deploys it from the PR that is being tested.

Rook v1.18.4 works well with Ceph Squid (v19). But for NVMe-oF testing we need a Ceph Tentacle (v20) cluster.

Closes: #5772

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

* `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)

</details>

nixpanic committed:
ci: use Rook 1.19.3

Recent versions of Rook use Ceph-CSI Operator by default. There is no
need to have Ceph-CSI deployed by Rook, as our CI deploys it from the PR
that is being tested.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.34

nixpanic committed:
ci: let Rook deploy a Ceph Tentacle cluster

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic commented:
Rook 1.18.4 with Ceph v19: https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/mini-e2e_k8s-1.34/detail/mini-e2e_k8s-1.34/84/pipeline

Rook 1.18.4 with Ceph Tentacle: https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/mini-e2e_k8s-1.34/detail/mini-e2e_k8s-1.34/85/pipeline

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.33

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.32

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.32

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.33/cephfs

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.33/cephfs

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.33/cephfs

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.33/cephfs

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.33/cephfs

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.32/cephfs

nixpanic commented:
> /test ci/centos/mini-e2e-helm/k8s-1.32/cephfs

Using the image (manually pushed into the CI registry) from https://tracker.ceph.com/issues/73567

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.32/cephfs

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.34/cephfs

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.34/cephfs

nixpanic commented:
All three CI jobs failed with the following error:
```
failed to append data: could not append to file /var/lib/www/test: failed to execute command: command terminated with exit code 1 ; stderr: /bin/sh: 1: echo: echo: I/O error
```

Added a note about this to #5772.

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34/rbd

Madhu-1 commented:
@nixpanic CI is still failing?

nixpanic committed:
e2e: enable fencing for operator deployment

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34/cephfs

nixpanic commented:
> @nixpanic CI is still failing?

Yeah, there is an other test that fails on CephFS too...

nixpanic commented:
/retest ci/centos/k8s-e2e-external-storage/1.34

nixpanic commented:
> /retest ci/centos/k8s-e2e-external-storage/1.34

Rook deployment got stuck and timed out.

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34

nixpanic commented:
/test ci/centos/k8s-e2e-external-storage/1.34

nixpanic commented:
**mini-e2e-operator_k8s-1.34** fails with strange errors, NFS can not provision a volume:
```
  I1120 15:34:45.687189 77525 pvc.go:85] PVC cephcsi-nfs-pvc Event: ProvisioningFailed - Volume provisioning failed with infeasible error. Retries will be delayed. rpc error: code = InvalidArgument desc = failed to create export: failed to create export "0001-0024-2f9ad70c-be67-40ee-9ded-8523010c4847-0000000000000001-0acfc816-a645-4c71-bfd5-0f27c931994b" in NFS-cluster "my-nfs"(an error (exit status 22) occurred while running ceph args: [--id cephcsi-cephfs-provisioner --keyfile=***stripped*** -m rook-ceph-mon-a.rook-ceph.svc.cluster.local:6789 nfs export create cephfs myfs my-nfs /0001-0024-2f9ad70c-be67-40ee-9ded-8523010c4847-0000000000000001-0acfc816-a645-4c71-bfd5-0f27c931994b /volumes/csi/nfs-export-0acfc816-a645-4c71-bfd5-0f27c931994b/c12f4fa8-32cb-437e-b18d-366819dd7c0a]): Error EINVAL: Traceback (most recent call last):
    File "/usr/share/ceph/mgr/mgr_module.py", line 2010, in _handle_command
      return CLICommand.COMMANDS[cmd['prefix']].call(self, cmd, inbuf)
    File "/usr/share/ceph/mgr/mgr_module.py", line 534, in call
      return self.func(mgr, **kwargs)
    File "/usr/share/ceph/mgr/object_format.py", line 592, in _format_response
      robj = f(*args, **kwargs)
    File "/usr/share/ceph/mgr/nfs/module.py", line 45, in _cmd_nfs_export_create_cephfs
      earmark_resolver = CephFSEarmarkResolver(self)
    File "/usr/share/ceph/mgr/mgr_util.py", line 356, in __init__
      self._cephfs_client = client or CephfsClient(mgr)
    File "/usr/share/ceph/mgr/mgr_util.py", line 318, in __init__
      self.connection_pool = CephfsConnectionPool(self.mgr)
    File "/usr/share/ceph/mgr/mgr_util.py", line 223, in __init__
      self.timer_task.start()
    File "/usr/lib64/python3.9/threading.py", line 899, in start
      _start_new_thread(self._bootstrap, ())
  RuntimeError: can't start new thread
```

[logs](https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/mini-e2e-operator_k8s-1.34/detail/mini-e2e-operator_k8s-1.34/2/pipeline/102)

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.34

nixpanic committed:
nfs: log error when go-ceph fails to create the export

With Ceph Tentacle creating a (the first?) NFS-export fails with:

```
Error EINVAL: Traceback (most recent call last):
    File "/usr/share/ceph/mgr/mgr_module.py", line 2010, in _handle_command
      return CLICommand.COMMANDS[cmd['prefix']].call(self, cmd, inbuf)
    File "/usr/share/ceph/mgr/mgr_module.py", line 534, in call
      return self.func(mgr, **kwargs)
    File "/usr/share/ceph/mgr/object_format.py", line 592, in _format_response
      robj = f(*args, **kwargs)
    File "/usr/share/ceph/mgr/nfs/module.py", line 45, in
_cmd_nfs_export_create_cephfs
      earmark_resolver = CephFSEarmarkResolver(self)
    File "/usr/share/ceph/mgr/mgr_util.py", line 356, in __init__
      self._cephfs_client = client or CephfsClient(mgr)
    File "/usr/share/ceph/mgr/mgr_util.py", line 318, in __init__
      self.connection_pool = CephfsConnectionPool(self.mgr)
    File "/usr/share/ceph/mgr/mgr_util.py", line 223, in __init__
      self.timer_task.start()
    File "/usr/lib64/python3.9/threading.py", line 899, in start
      _start_new_thread(self._bootstrap, ())
  RuntimeError: can't start new thread
```

It seems this can only happen when the CLI is used, that means go-ceph
failed to create the export in the first place. It would be useful to
log the go-ceph failure, so that improving is possible.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34/nfs

nixpanic committed:
e2e: wait for operator deployment to disable metadata

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34

nixpanic commented:
**ci/centos/mini-e2e-operator/k8s-1.34** didn't really fail, it timed out after 2 hours of e2e

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34

nixpanic committed:
e2e: restart CSI pods after changing metadata config option

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34/rbd

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34/cephfs

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34/nfs

Rakshith-R commented:
Hey @nixpanic 
Can you please try tentacle ci with Read affinity disabled ?
https://github.com/ceph/ceph-csi/pull/5857/files#diff-79a327c64853a938465ec1bbc654be68650c4503d26a083edac7d91f968393bf

nixpanic commented:
@Rakshith-R , what is the reason you think read affinity causes the issue?

nixpanic committed:
e2e: report an error if writing persistent data fails

Use `dd` to sync the data, do not wait for unmounting to flush
outstanding I/O (and ignore failed writes).

See-also: https://tracker.ceph.com/issues/73997?#note-27
Co-authored-by: Patrick Donnelly <pdonnell@ibm.com>
Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34/cephfs

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34/cephfs

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.34/cephfs

bumarcell commented:
I believe support for Tentacle was added in Rook [v1.18.8](https://github.com/rook/rook/releases/tag/v1.18.8)

nixpanic commented:
> I believe support for Tentacle was added in Rook [v1.18.8](https://github.com/rook/rook/releases/tag/v1.18.8)

It is not so much an issue with Rook, more with a regression in Tentacle with CephFS. We're waiting for a Tentacle release that includes ceph/ceph#66611.

bumarcell commented:
Got it 👍 
That said, is there a reason to prefer upgrading to v1.18.4 rather than 1.18.8 or 1.18.9?

nixpanic commented:
This PR has been open for a while already. I guess at the time it was opened, Rook 1.18.4 was the most recent release. When we update this PR, we definitely should update the Rook version to whatever is current. Thanks for the reminder!

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.35/rbd

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.35/rbd

nixpanic commented:
@mergifyio rebase

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.35

nixpanic commented:
/test ci/centos/mini-e2e-operator/k8s-1.35

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.35

nixpanic commented:
[ci/centos/mini-e2e-helm/k8s-1.35](https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/job/mini-e2e-helm_k8s-1.35/159/display/redirect) passed the CephFS tests that failed on previous Ceph Tentacle releases.

iPraveenParihar commented:
@mergifyio rebase
